From 45e40867e08e1ad6e3666bd0efb8d9c9d5ff20bf Mon Sep 17 00:00:00 2001
From: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Date: Mon, 10 Dec 2018 15:37:21 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399-tve1030g: distinguish isp relate
 nodes for Android 9.0 & 8.1

rk3399-tve1030g*:
    isp0/1 for CameraHal1 in Android8.1 or lower version
    rkisp1_0/1 & mipi_dphy_rx0/tx1rx1 for CameraHal3 in Android9.0

Change-Id: I9198eccb0d76ad5e51ff6a2a5021e5acefbf0f49
Signed-off-by: Wang Panzhenzhuan <randy.wang@rock-chips.com>
---
 .../boot/dts/rockchip/rk3399-tve1030g-avb.dts | 143 ++++++++++++++++++
 .../boot/dts/rockchip/rk3399-tve1030g.dts     |   7 +
 .../boot/dts/rockchip/rk3399-tve1030g.dtsi    |   8 -
 3 files changed, 150 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g-avb.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g-avb.dts
index 9c0b90c3cbaa..b99cfc4a3709 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g-avb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g-avb.dts
@@ -13,6 +13,7 @@
 	};
 
 };
+
 &firmware_android {
 	compatible = "android,firmware";
 	boot_devices = "fe330000.sdhci";
@@ -31,3 +32,145 @@
 		};
 	};
 };
+
+&i2c1 {
+	status = "okay";
+
+	ov13850: ov13850@10 {
+		compatible = "ovti,ov13850";
+		status = "disabled";
+		reg = <0x10>;
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "xvclk";
+		/* avdd-supply = <>; */
+		/* dvdd-supply = <>; */
+		/* dovdd-supply = <>; */
+		/* reset-gpios = <>; */
+		reset-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+		pwdn-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "rockchip,camera_default";
+		pinctrl-0 = <&cif_clkout>;
+		port {
+			ucam_out0: endpoint {
+				remote-endpoint = <&mipi_in_ucam0>;
+				//remote-endpoint = <&mipi_in_ucam1>;
+				data-lanes = <1 2>;
+			};
+		};
+	};
+
+	gc2355: gc2355@3c {
+		status = "okay";
+		compatible = "galaxycore,gc2355";
+		reg = <0x3c>;
+		clocks = <&cru SCLK_CIF_OUT>;
+		clock-names = "xvclk";
+		/* avdd-supply = <>; */
+		/* dvdd-supply = <>; */
+		/* dovdd-supply = <>; */
+		/* reset-gpios = <>; */
+		pwdn-gpios = <&gpio2 12 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "rockchip,camera_default";
+		pinctrl-0 = <&cif_clkout>;
+		port {
+			ucam_out1: endpoint {
+				//remote-endpoint = <&mipi_in_ucam0>;
+				remote-endpoint = <&mipi_in_ucam1>;
+				data-lanes = <1>;
+			};
+		};
+	};
+};
+
+&mipi_dphy_rx0 {
+	status = "disabled";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam0: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&ucam_out0>;
+				data-lanes = <1 2>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_rx0_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&isp0_mipi_in>;
+			};
+		};
+	};
+};
+
+&mipi_dphy_tx1rx1 {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			mipi_in_ucam1: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&ucam_out1>;
+				data-lanes = <1>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_tx1rx1_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&isp1_mipi_in>;
+			};
+		};
+	};
+};
+
+&rkisp1_0 {
+	status = "disabled";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp0_mipi_in: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dphy_rx0_out>;
+		};
+	};
+};
+
+&rkisp1_1 {
+	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		isp1_mipi_in: endpoint@0 {
+			reg = <0>;
+			remote-endpoint = <&dphy_tx1rx1_out>;
+		};
+	};
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dts b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dts
index a59dfc6ad7fb..28f81ee8f7f5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dts
@@ -33,3 +33,10 @@
 	};
 };
 
+&isp0 {
+	status = "okay";
+};
+
+&isp1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
index b9baa5c6abde..7005f22f7edf 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tve1030g.dtsi
@@ -707,14 +707,6 @@
 	gpio1830-supply = <&vcc_3v0>;
 };
 
-&isp0 {
-	status = "okay";
-};
-
-&isp1 {
-	status = "okay";
-};
-
 &isp0_mmu {
 	status = "okay";
 };
-- 
2.35.3

